Src/DependencyCollector/NetCore/DependencyCollectorDiagnosticListener.cs

```diff
@@ -82,129 +100,247 @@ public void OnError(Exception error)
         }
 
         /// <summary>
-        /// Get the DependencyTelemetry objects that are still waiting for a response from the dependency. This will most likely only be used for testing purposes.
+        /// Enables HttpClient instrumentation with activity. It is never called and allows Microsoft.Extensions.DiagnosticAdapter to know that event is enabled.
         /// </summary>
-        internal IEnumerable<DependencyTelemetry> PendingDependencyTelemetry
+        [DiagnosticName("System.Net.Http.HttpRequestOut")]
+        public void IsActivityEnabled()
         {
-            get { return pendingTelemetry.Values; }
         }
 
         /// <summary>
-        /// Diagnostic event handler method for 'System.Net.Http.Request' event.
+        /// Handler for Exception event, it is sent when request processing cause an exception (e.g. because of DNS or network issues) on netcoreapp2.0 only
+        /// Stop event will be sent anyway with null response.
         /// </summary>
-        [DiagnosticName("System.Net.Http.Request")]
-        public void OnRequest(HttpRequestMessage request, Guid loggingRequestId)
+        [DiagnosticName("System.Net.Http.Exception")]
+        public void OnException(Exception exception, HttpRequestMessage request)
         {
-            try
+            Activity currentActivity = Activity.Current;
+            if (currentActivity == null)
+            {
+                DependencyCollectorEventSource.Log.CurrentActivityIsNull();
+                return;
+            }
+
+            this.pendingExceptions.TryAdd(currentActivity.Id, exception);
+            this.client.TrackException(exception);
```

Comment:
the Exception event is only fired for the exceptions happened during Http request processing. So if the exception is fired in TrackException, it will not cause another Exception event from Http DiganoticSource